### PR TITLE
feat : Add full analytics module (events, metrics, tasks, views, admin, tests)

### PR DESCRIPTION
**Pull request by Surya443**
- Implemented full analytics system with event tracking, daily/monthly metrics, feature usage metrics, and optimized query indexing.
- Created complete event ingestion workflow with TrackEventSerializer, services-based processing, and public tracking endpoint.
- Added metrics aggregation services for DAU/WAU/MAU, revenue computation, and feature usage ranking.
Implemented Celery tasks for daily and monthly metric aggregation with Celery Beat scheduling (feature aggregation optional for now)
- Created analytics API endpoints (events list, metrics, dashboard) with filtering, date-range queries, and viewsets
-Added 11 comprehensive tests (offline testing with no real API Calls) for services and views, covering event tracking, metrics calculations, revenue aggregation, and API responses

### CODE REVIEW

File: analytics/tasks.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# analytics/tasks.py
+import datetime
+from celery import shared_task
+from django.db.models import Count, Sum
```
Comment (srivatsan0611) [inline analytics/tasks.py]: @Surya443 - we're missing `Min` import along with Count & Sum lol , IDE showed that. Please add the `Min` import

File: analytics/tasks.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+from django.utils import timezone
+from organizations.models import Organization
+from .models import Event, DailyMetric, MonthlyMetric, FeatureMetric
+
+
+@shared_task(bind=True, max_retries=3, default_retry_delay=60)
+def aggregate_daily_metrics(self):
```
Comment (srivatsan0611) [inline analytics/tasks.py]: Nice work on the celery tasks. Can we also add these as "Celery Beat Schedule" to `config/settings.py`?

**Review by srivatsan0611 — COMMENTED** [review]
Hi @Surya443 - thanks for the PR da. 

All the MVTs for Analytics look great. I have left some minor P3 comments on tasks.py alone. Rest all look well written, nice work da thanks.

I just thought test coverage could be expanded - Only 9 tests total. Consider adding:
- Admin tests
- Celery task tests
- Serializer validation tests
- Edge cases (empty data, multi-org isolation)

This may help us both in identifying hidden bugs iteratively so that we can avoid manual mistakes. 
Try giving that to GPT/Claude to expand tests to these sections individually (you can give your `serailizers.py` and ask gpt to generate a `test_serializers.py` for instance).

**Review by srivatsan0611 — APPROVED** [review]
mass machi
